### ISSUE TITLE
dayjs('2020-10-01').add(dayjs.duration(1, 'months')) not returns dayjs('2020-11-01')

### ISSUE THREAD

**Issue by chimerast** [iamkun/dayjs#1515]
**Describe the bug**
`dayjs.duration(1, 'months')` is converted to `dayjs.duration(30, 'days')` internally by duration plugin.
so `dayjs('2020-10-01').add(dayjs.duration(1, 'months'))` returns `dayjs('2020-10-31')`

**Expected behavior**
`dayjs('2020-10-01').add(dayjs.duration(1, 'months'))` should returns `dayjs('2020-11-01')`

**Information**
 - Day.js Version: 1.10.5
 - OS: macOS 11.4
 - Browser: Chrome 91
 - Time zone: GMT+09:00 JST

**Comment by chimerast** [comment]
### To reproduce

```diff
diff --git a/test/plugin/duration.test.js b/test/plugin/duration.test.js
index cca1a82..57456af 100644
--- a/test/plugin/duration.test.js
+++ b/test/plugin/duration.test.js
@@ -174,8 +174,8 @@ describe('Add', () => {

 test('Add duration', () => {
   const a = dayjs('2020-10-01')
-  const days = dayjs.duration(2, 'days')
-  expect(a.add(days).format('YYYY-MM-DD')).toBe('2020-10-03')
+  const days = dayjs.duration(1, 'months')
+  expect(a.add(days).format('YYYY-MM-DD')).toBe('2020-11-01')
 })

 describe('Subtract', () => {
 ```

### Results
```
  ● Add duration

    expect(received).toBe(expected) // Object.is equality

    Expected value to be:
      "2020-11-01"
    Received:
      "2020-10-31"

      176 |   const a = dayjs('2020-10-01')
      177 |   const days = dayjs.duration(1, 'months')
    > 178 |   expect(a.add(days).format('YYYY-MM-DD')).toBe('2020-11-01')
      179 | })
      180 |
      181 | describe('Subtract', () => {

      at Object.<anonymous> (test/plugin/duration.test.js:178:44)
 ```

**Comment by EPillin** [comment]
Having recently encountered this issue too, I also think this deserves a fix.

While it can be considered acceptable to say that a month is 30 days in general, a specific point of reference is provided in the context of the `dayjs.add( dayjs.duration )` function. In this context, it is legitimate to expect the library to infer the exact duration of this month, which may or may not be 30 days.

Not only is the current behaviour incorrect, but it is inconsistent with that of `.add(value, unit)`, since `date.add( dayjs.duration({ months: 1 }) )` can be different from `date.add(1, 'months')`.

The latter also proves that it is possible for this library to handle this situation properly, by handling durations in the context of additions and subtractions in terms of each of their "components" (duration in years, months, days, ...) instead of being converted to milliseconds at [this line](https://github.com/iamkun/dayjs/blob/dev/src/plugin/duration/index.js#L268).

**Comment by Magnuti** [comment]
I don't think this issue is relevant any longer. Both of these two tests succeeds on my machine.

```ts
dayjs.tz.setDefault("Asia/Tokyo");
assert.equal(dayjs("2020-10-01").add(1, "month").isSame(dayjs("2020-11-01"), "day"), true);
assert.equal(dayjs("2020-10-01").add(dayjs.duration(1, "months")).isSame(dayjs("2020-11-01"), "day"), true);
```
